### PR TITLE
Fix SMF tax-based payments

### LINKED ISSUE #424: SMF payments include non-taxpayers

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Fixes #424

### CODE REVIEW

**Review by MaxGhenis — APPROVED** [review]

File: policyengine/countries/uk/default_reform/reform.py [hunk 1]
```diff
@@ -619,8 +619,10 @@ def formula(household, period, parameters):
             person = household.members
             tax_band = person("tax_band", period)
             bands = tax_band.possible_values
-            eligible = ~household.any(
-                (tax_band == bands.ADDITIONAL) | (tax_band == bands.HIGHER)
+            eligible = household.any(
+                (tax_band == bands.BASIC)
+                | (tax_band == bands.STARTER)
+                | (tax_band == bands.INTERMEDIATE)
             ) * (household("smf_benefit_cash_payment", period) == 0)
```
Comment (MaxGhenis) [inline policyengine/countries/uk/default_reform/reform.py:626]: Actually could we change this logic so it works even with there's no benefit parameter set? That is, use eligibility for it.

Comment (nikhilwoodruff) [reply policyengine/countries/uk/default_reform/reform.py:626]: Yep on it

**Review by MaxGhenis — COMMENTED** [review]
Could you also add unit tests? Or is that not possible since it's outside openfisca?

**Comment by nikhilwoodruff** [comment]
Turned out to be unexpectedly complex:
* openfisca testing CLI doesn't handle our reform tuples
* reform object returns a function and needs instantiation (got this but couldn't solve above)

File: policyengine/countries/uk/default_reform/reform.py [hunk 2]
```diff
@@ -602,8 +598,38 @@ def formula(household, period, parameters):
...
+        def formula(household, period):
+            person = household.members
+            tax_band = person("tax_band", period)
+            bands = tax_band.possible_values
+
+            return household.any(
+                (tax_band == bands.BASIC)
+                | (tax_band == bands.STARTER)
+                | (tax_band == bands.INTERMEDIATE)
+            ) * ~household("smf_benefit_payment_eligible", period)
```
Comment (MaxGhenis) [inline policyengine/countries/uk/default_reform/reform.py:617]:
```suggestion
            ) & ~household("smf_benefit_payment_eligible", period)
```

File: policyengine/countries/uk/default_reform/reform.py [hunk 3]
```diff
@@ -602,8 +598,38 @@ def formula(household, period, parameters):
...
+
+    class smf_benefit_cash_payment(Variable):
+        entity = Household
+        definition_period = YEAR
+        label = "Benefit-based cash payment"
```
Comment (MaxGhenis) [inline policyengine/countries/uk/default_reform/reform.py:622]:
```suggestion
        label = "SMF benefit-based cash payment"
```

File: policyengine/countries/uk/default_reform/reform.py [hunk 4]
```diff
@@ -602,8 +598,38 @@ def formula(household, period, parameters):
...
+        documentation = (
+            "The benefit-based cash payment this household receives."
```
Comment (MaxGhenis) [inline policyengine/countries/uk/default_reform/reform.py:624]: Similar for the tax one
```suggestion
            "The Social Market Foundation's benefit-based cash payment this household receives."
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**